diff -Nrup a/setup.py b/setup.py
--- a/setup.py	2019-04-19 18:23:56.000000000 +0200
+++ b/setup.py	2019-10-09 16:53:05.666163481 +0200
@@ -8,13 +8,13 @@ if sys.platform == "win32":
 
 
 sources = ['c/_cffi_backend.c']
-libraries = ['ffi']
+libraries = ['ffi', 'dl']
 include_dirs = ['/usr/include/ffi',
                 '/usr/include/libffi']    # may be changed by pkg-config
 define_macros = []
 library_dirs = []
 extra_compile_args = []
-extra_link_args = []
+extra_link_args = ['-ldl']
 
 
 def _ask_pkg_config(resultlist, option, result_prefix='', sysroot=False):
@@ -220,7 +220,7 @@ Contact
             define_macros=define_macros,
             library_dirs=library_dirs,
             extra_compile_args=extra_compile_args,
-            extra_link_args=extra_link_args,
+            extra_link_args=['-lffi', '-ldl'],
         )] if cpython else [],
 
         install_requires=[
